### DISCUSSION TITLE
Low Camera Framerate

### DISCUSSION THREAD

**Discussion by Grea5eMonkey** [xblax/flashforge_ad5m_klipper_mod#125] [Q&A] [answered]
I just installed the add-on camera module from FlashForge on my AD5M and getting very low framerate (1-2 FPS). Using the adaptive mjpeg option on Mainsail. Any suggestions on getting a better framerate, or is this just a limitation of processing power on the main board? On port 8080/status it's reporting 1280x720 resolution but I don't see any config files I can modify to try lowering it. Thanks!

**Comment by consp** [comment]
> or is this just a limitation of processing power on the main board 

That would be my guess, does it actually support mjpeg in hardware (as in the camera)? Maybe try different encoding options?

~~1-2fps would be my guess the hardware supports encoding in software 720p.~~ (thought it needed too much memory for it to be sustainable, apperantly it doesnt)

**Comment by xblax** [comment, marked as answer]
The Mainboard can easily support the full framerate. I don't think that's the issue. I had this problem when the the Wifi connection was very bad. Either high latency or low throughput (leading to high latency).

You can try if the 'UV4L-MJPEG' option works better, but there might be some build-up of delay.

**Comment by Grea5eMonkey** [comment]
Update to this. I replaced the printer's WiFi antenna with an external antenna from Amazon, hoping to get a little better reception in my garage where the printer lives. Using the UV4L-MJPG option before and after, there's still some delay but it's significantly less. Unless there are other factors I'm not considering and it's just a placebo effect! <s>I did notice a decrease in ping time. ~100-160ms before the new antenna, and now hovering around 60ms with the new antenna. </s> I'll also note that using the Adaptive MJPEG option with framerate displayed it was reporting the same as before, around 2 FPS.
